3.Tests/UI/StepDefinitions/Project/PostStepDefinition.cs

```diff
@@ -1,41 +1,70 @@
 ﻿using System;
+using SeleniumExtras.WaitHelpers;
 using TechTalk.SpecFlow;
+using Todoly.Core.Helpers;
+using Todoly.Core.UIElements.Drivers;
 using Todoly.Tests.UI.Steps.Commons;
+using Todoly.Views.WebAppPages;
 
 namespace SeleniumTest.Tests.Steps.Project;
 
 [Binding]
 [Scope(Feature = "Project Creation")]
 public class PostStepDefinitions : CommonSteps
 {
+    private readonly HomePage _homePage;
     private readonly ScenarioContext _scenarioContext;
 
     public PostStepDefinitions(ScenarioContext scenarioContext) : base(scenarioContext)
     {
+        _homePage = new HomePage();
         _scenarioContext = scenarioContext;
     }
 
     [When(@"the user clicks the New Project button")]
-    public void WhenTheUserClicksTheNewProjectButton()
+    public void ClickNewProjectButton()
     {
-        _scenarioContext.Pending();
+        _homePage.AddNewProjectButton.Click();
     }
 
     [When(@"inputs a new project name")]
-    public void WhenInputsANewProjectName()
+    public void InputsNewProjectName()
     {
-        _scenarioContext.Pending();
+        string projectName = IdHelper.GetNewId();
+
+        _homePage.AddNewProjectInput.Clear();
+        _homePage.AddNewProjectInput.Type(projectName);
+
+        _scenarioContext[ConfigModel.CurrentProject] = projectName;
     }
 
     [When(@"clicks the Add button")]
-    public void WhenClicksTheAddButton()
+    public void ClicksTheAddButton()
     {
-        _scenarioContext.Pending();
+        _homePage.AddNewProjectNameButton.Click();
     }
 
     [Then(@"a new project with the chosen name should be displayed in the projects list")]
-    public void ThenANewProjectWithTheChosenNameShouldBeDisplayedInTheProjectsList()
+    public void VerifyProjectCreation()
     {
-        _scenarioContext.Pending();
+        _scenarioContext.TryGetValue(ConfigModel.CurrentProject, out string projectName);
+
+        GenericWebDriver.Wait.Until(
+            ExpectedConditions.TextToBePresentInElement(
+                _homePage.ProjectTitleDiv.WebElement,
+                projectName
+            )
+        );
+
+        string actualText = _homePage.GetProjectTd(projectName).WebElement.Text;
+
+        try
+        {
+            Assert.That(projectName, Is.EqualTo(actualText));
+        }
+        catch (Exception)
+        {
+            throw new Exception("Error: Project was not successfully created");
```

Comment:
We should not throw exceptions on steps, not even Then steps, Exception are better to be thrown on base WebElement classes or API helpers

Reply:
Noted.